zulipterminal/ui_tools/buttons.py

```diff
@@ -322,14 +322,20 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
         return super().keypress(size, key)
 
 
-class DecodedStream(TypedDict):
-    stream_id: Optional[int]
-    stream_name: Optional[str]
+class DecodedStreamMeta(TypedDict, total=False):
+    stream_id: int
+    stream_name: str
+
+
+class DecodedStreamPatched(TypedDict):
```

Comment:
How about `DecodedStreamPartial` and `DecodedStreamAbsolute`?

Reply:
Sounds good :+1